src/Auth/Models/ErrorViewModel.cs

```diff
@@ -1,10 +1,9 @@
 ﻿// Copyright (c) Brock Allen & Dominick Baier. All rights reserved.
```

Comment:
...

Reply:
I asked in discord @ArtOfCode- and he said its ok. I personally am not sure what to do about these

Reply:
https://tldrlegal.com/license/apache-license-2.0-(apache-2.0)

Reply:
Copyright belongs to whoever wrote the code, regardless of whether this notice is present in the source code or not.

It also applies only to the code originally written by these authors.

So, keeping the copyright notice is not only "ok", but also the sane thing to do.

Reply:
How much would the code have to change for the licensing to be removed?

Reply:
As long as there is one line of code written by that person, the copyright notice is due. And I do believe that even the file name by itself also counts. Perhaps @ArtOfCode- can confirm.

In any case: What we're seeing here is a *copyright notice*, not a license statement.

When releasing software, the copyright notice serves to identify the author who made it available to the public. This particular copyright notice belongs to Identity Server 4, which is released under the Apache 2.0 license. We, and future contributors, *should* know who licensed these files under the terms of that license, for various reasons. The notice should remain there. Legally, it would make no difference; however, it's an actual requirement of the Apache 2.0 license to have the notice in every file distributed under it. (see: https://opensource.stackexchange.com/q/6909/3988)

"All rights reserved" means, mainly, two things: (1) the author(s) is free to release newer versions of the software under a different license of their choice; and (2) they have the right to sue anyone who infringes their license terms (which is expected indeed).

Furthermore, I believe contributors can add their own copyright notices in addition to the one(s) already there. Can't state that for sure, though. More info needed.

Reply:
"All rights reserved" is not contradictory to a license grant. It's not a _necessary_ statement, but it's also not wrong. Keeping these license headers in is the generally accepted way to go.

Reply:
Marking as resolved but feel free to bring up any additional concerns on Discord or in a new issue.